src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SwaggerSamClient.scala

```diff
@@ -0,0 +1,128 @@
+    apiClient.setAccessToken(accessToken)
+    apiClient.setBasePath(samBasePath)
+    new GoogleApi(apiClient)
+  }
+
+  //A resources API if you already have a token
+  private[auth] def samResourcesApi(accessToken: String): ResourcesApi = {
+    val apiClient = new ApiClient()
+    apiClient.setAccessToken(accessToken)
+    apiClient.setBasePath(samBasePath)
+    new ResourcesApi(apiClient)
+  }
+
+  //A resources API as the given user's pet SA
+  private[auth] def resourcesApiAsPet(userEmail: WorkbenchEmail, googleProject: GoogleProject): ResourcesApi = {
+    samResourcesApi(getCachedPetAccessToken(userEmail, googleProject))
+  }
+
+  //"Fast" lookup of pet's access token, using the cache.
+  private def getCachedPetAccessToken(userEmail: WorkbenchEmail, googleProject: GoogleProject): String = {
+    petTokenCache.get(UserEmailAndProject(userEmail, googleProject))
+  }
+
+  private[leonardo] val petTokenCache = CacheBuilder.newBuilder()
+    .expireAfterWrite(cacheExpiryTime, TimeUnit.MINUTES)
+    .maximumSize(cacheMaxSize)
+    .build(
+      new CacheLoader[UserEmailAndProject, String] {
+        def load(userEmailAndProject: UserEmailAndProject) = {
+          getPetAccessTokenFromSam(userEmailAndProject.userEmail, userEmailAndProject.googleProject)
+        }
+      }
+    )
+
+  //"Slow" lookup of pet's access token. The cache calls this when it needs to.
+  private def getPetAccessTokenFromSam(userEmail: WorkbenchEmail, googleProject: GoogleProject): String = {
+    val samAPI = samGoogleApi(getAccessTokenUsingPem(leoEmail, leoPem))
+    val userPetServiceAccountKey = samAPI.getUserPetServiceAccountKey(googleProject.value, userEmail.value)
+    val keyTreeMap = userPetServiceAccountKey.asInstanceOf[LinkedTreeMap[String,String]]
+    getAccessTokenUsingJson(new Gson().toJsonTree(keyTreeMap).toString)
+  }
+
+  //Given a pem, gets an access token
+  private def getAccessTokenUsingPem(email: WorkbenchEmail, pem: File): String = {
+    val credential = new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(email.value)
+      .setServiceAccountScopes(saScopes.asJava)
+      .setServiceAccountPrivateKeyFromPemFile(pem)
+      .build()
+
+    credential.refreshToken
+    credential.getAccessToken
+  }
+
+
+  //Given some JSON, gets an access token
+  private def getAccessTokenUsingJson(saKey: String) : String = {
+    val keyStream = new ByteArrayInputStream(saKey.getBytes)
+    val credential = ServiceAccountCredentials.fromStream(keyStream).createScoped(saScopes.asJava)
+    credential.refreshAccessToken.getTokenValue
+  }
+
+
+  private def getClusterResourceId(googleProject: GoogleProject, clusterName: ClusterName): String = {
+    googleProject.value + "_" + clusterName
+  }
+
+
+  def createNotebookClusterResource(userEmail: WorkbenchEmail, googleProject: GoogleProject, clusterName: ClusterName)(implicit executionContext: ExecutionContext) = {
```

Comment:
I don't think any of these methods need to take an implicit `ExecutionContext` -- nothing in this class is dealing with futures.